wormhole-connect/src/theme.ts

```diff
@@ -300,15 +323,47 @@ export const dark: WormholeConnectTheme = {
   logo: '#ffffff',
 };
 
-export const getDesignTokens = (customTheme: WormholeConnectPartialTheme) => {
-  const baseTheme = customTheme?.mode === 'light' ? light : dark;
-  const theme = Object.assign(
-    {},
-    baseTheme,
-    customTheme,
-  ) as WormholeConnectTheme;
+export const generateTheme = (customTheme: WormholeConnectTheme): Theme => {
+  const baseTheme = customTheme.mode === 'light' ? light : dark;
+  const theme = Object.assign({}, baseTheme) as InternalTheme;
 
-  theme.background.default = 'transparent';
+  // Override built-in theme with whichever custom values we've been provided
+  if (customTheme) {
+    if (customTheme.input) {
+      theme.modal = {
+        background: customTheme.input,
+      };
+    }
+    if (customTheme.primary) {
+      theme.primary = {
+        main: customTheme.primary,
```

Comment:
MUI derives all the other values for `PaletteColor` when just provided with `main`.